scss/components/_menu.scss

```diff
@@ -34,40 +38,29 @@ $menu-icon-spacing: 0.25rem !default;
 /// @type Color
 $menu-item-background-hover: $light-gray !default;
 
-/// Color for bordered menu
-/// @type Color
-$menu-border: $light-gray !default;
-
 /// Creates the base styles for a Menu.
 @mixin menu-base {
-  margin: $menu-margin;
-  list-style-type: none;
+  padding: 0;
+  margin: 0;
+  list-style: none;
+  position: relative;
 
   @if $global-flexbox {
     display: flex;
     flex-wrap: nowrap;
-    align-items: center;
-    width: 100%;
   }
 
-  // List items are table cell to allow for vertical alignment
-  > li {
+  li {
     @include disable-mouse-outline;
-
-    @if $global-flexbox {
-      flex: 0 0 auto;
-    }
-    @else {
-      display: table-cell;
-      vertical-align: middle;
-    }
   }
 
-  // Reset line height to make the height of the overall item easier to calculate
-  > li > a {
-    display: block;
-    padding: $menu-item-padding;
+  a,
+  .button {
```

Comment:
I think this is carried over fro the old menu system. You are right though, it maybe a bit heavy handed. Having said that, I get why its there - a mini reset. I think `text-decoration` could come out, as thats definately not needed (it should respect whatever settings you have). We could try removing it and seeing how this effects an example including buttons, but I think its needed as we dont use a class on our `a` links within menus to style them (and we needed to keep backwards compatibility).